File: golang/array_string/romantoint.go
package main

func romanToInt(s string) int {
	roman_to_int := map[byte]int{
		'I': 1,
		'V': 5,
		'X': 10,
		'L': 50,
		'C': 100,
		'D': 500,
		'M': 1000,
	}

	total := 0

	for i := 0; i < len(s); i++ {
		if i < len(s)-1 && roman_to_int[s[i]] < roman_to_int[s[i+1]] {
			total -= roman_to_int[s[i]]
		} else {
			total += roman_to_int[s[i]]
		}
	}

	return total
}

File: java/src/main/java/com/sanedge/interview/array_string/RomanToInt.java
package com.sanedge.interview.array_string;

import java.util.HashMap;
import java.util.Map;

public class RomanToInt {

    public int romanToInt(String s) {
        Map<Character, Integer> romanToInt = new HashMap<>();
        romanToInt.put('I', 1);
        romanToInt.put('V', 5);
        romanToInt.put('X', 10);
        romanToInt.put('L', 50);
        romanToInt.put('C', 100);
        romanToInt.put('D', 500);
        romanToInt.put('M', 1000);

        int total = 0;

        for (int i = 0; i < s.length(); i++) {
            if (
                i < s.length() - 1 &&
                romanToInt.get(s.charAt(i)) < romanToInt.get(s.charAt(i + 1))
            ) {
                total -= romanToInt.get(s.charAt(i));
            } else {
                total += romanToInt.get(s.charAt(i));
            }
        }

        return total;
    }
}
